libcontainer/nsenter/nsexec.c

```diff
@@ -184,18 +168,28 @@ static int write_file(char *data, size_t data_len, char *pathfmt, ...)
 	return ret;
 }
 
-#define SETGROUPS_ALLOW "allow"
-#define SETGROUPS_DENY  "deny"
+enum policy_t {
+	SETGROUPS_DEFAULT = 0,
+	SETGROUPS_ALLOW,
+	SETGROUPS_DENY,
+};
 
 /* This *must* be called before we touch gid_map. */
-static void update_setgroups(int pid, bool setgroup)
+static void update_setgroups(int pid, enum policy_t setgroup)
```

Comment:
i dont think we ever set this to deny or noop when we go into here ?, seems like something like `allow_setgroup(pid)` is simpler.